src/postgres.rs

```diff
@@ -0,0 +1,275 @@
+    client: Client,
+    insert_stmt: Statement,
+    table: String,
+}
+
+#[derive(Debug)]
+pub struct Postgres {
+    host: String,
+    port: u16,
+    user: String,
+    password: String,
+    db_name: String,
+    table: String,
+}
+
+impl Postgres {
+    pub fn new(
+        host: String,
+        port: u16,
+        user: String,
+        password: String,
+        db_name: String,
+        table: String,
+    ) -> Postgres {
+        Postgres {
+            host,
+            port,
+            user,
+            password,
+            db_name,
+            table,
+        }
+    }
+}
+
+#[async_trait]
+impl Db<PostgresConnection> for Postgres {
+    async fn connect(&self) -> Result<PostgresConnection> {
+        info!("Connecting to database");
+        let conn_str = format!(
+            "host={} port={} user={} password={} dbname={}",
+            self.host, self.port, self.user, self.password, self.db_name
+        );
+        let (client, connection) = tokio_postgres::connect(&conn_str, NoTls).await?;
+        tokio::spawn(Box::pin(connection));
+        info!("Successfully connected to database");
+
+        info!("Creating table {} if it does not exist", self.table);
+        let create_table_sql = format!(
+            r#"
+                CREATE TABLE IF NOT EXISTS {} (
+                    source text not null,
+                    semantics text not null,
+                    name text not null,
+                    seq bigint not null,
+                    psn bigint not null,
+                    timestamp bigint not null,
+                    payload jsonb,
+                    PRIMARY KEY (source, psn)
+                )
+            "#,
+            self.table
+        );
+        let create_table_statement = client.prepare(&*create_table_sql).await?;
+        client.execute(&create_table_statement, &[]).await?;
+        info!("Created table");
+
+        let sql = format!(
+            r#"INSERT INTO {} (source, semantics, name, seq, psn, timestamp, payload)
+                 SELECT * FROM unnest($1,$2,$3,$4,$5,$6,$7)
+                 ON CONFLICT DO NOTHING"#,
+            &self.table
+        );
+        let types = vec![
+            Type::TEXT_ARRAY,
+            Type::TEXT_ARRAY,
+            Type::TEXT_ARRAY,
+            Type::INT8_ARRAY,
+            Type::INT8_ARRAY,
+            Type::INT8_ARRAY,
+            Type::JSONB_ARRAY,
+        ];
+        let insert_stmt = client.prepare_typed(&sql, &types).await?;
+
+        let conn = PostgresConnection {
+            client,
+            insert_stmt,
+            table: self.table.clone(),
+        };
+
+        Ok(conn)
+    }
+}
+
+#[async_trait]
+impl DbConnection for PostgresConnection {
+    async fn insert(&self, items: Vec<Event<Payload>>) -> Result<()> {
+        let start = Instant::now();
+
+        let num_rows = items.len();
+        debug!("Preparing {} events", num_rows);
+        let rows: DbEventVec = (&items).into();
```

Comment:
```suggestion
        let rows = DbEventVec::from(&*items);
```
(would be my aesthetic preference)

Reply:
:+1: